test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/RemoteDependencyWalkerTests.cs

```diff
@@ -1669,7 +1669,144 @@ public async Task CentralTransitiveDependency_Downgrades_IfDowngrades_WereNotRej
             Assert.True(downgrade.DowngradedTo.Item.IsCentralTransitive);
             Assert.Equal("D", downgrade.DowngradedFrom.Key.Name);
             Assert.Equal("2.0.0", downgrade.DowngradedFrom.Key.VersionRange.OriginalString);
+        }
+
+        /// <summary>
+        ///   -> B 1.0.0 -> D [1.0.0] (PrivateAssets1)
+        /// A
+        ///   -> C 1.0.0 -> D [2.0.0] (PrivateAssets2)
+        /// </summary>
+        [Theory]
+        [InlineData(null, null, 1)]
+        [InlineData(LibraryIncludeFlags.All, null, 0)]
+        [InlineData(null, LibraryIncludeFlags.All, 0)]
+        [InlineData(LibraryIncludeFlags.All, LibraryIncludeFlags.All, 0)]
+        public async Task PrivateAssetsAll_VersionConflicts(LibraryIncludeFlags? privateAssets1,
+            LibraryIncludeFlags? privateAssets2, int expectedConflicts)
+        {
+            var framework = NuGetFramework.Parse("net45");
+            var context = new TestRemoteWalkContext();
+            var provider = new DependencyProvider();
+
+            provider.Package("A", "1.0.0")
+                .DependsOn("B", "1.0.0")
```

Comment:
The code allows is, but in practice B *has* to be a project.

Reply:
Yeah, I was thinking about that too, Unfortunately the `Test.Utility.DependencyProvider` doesn't let to mock  projects. On the other hand, projects and packages are handled equally by the graph walking algorithm so it doesn't really matter.

Reply:
Yep, it was mostly information for any future readers. The test is going through the intended codepath.